packages/functional-tests/tests/settings/totp.spec.ts

```diff
@@ -60,67 +63,119 @@ test.describe('severity-1 #smoke', () => {
 
     test('add TOTP and login', async ({
       target,
-      pages: { page, settings, totp, login },
+      pages: { page, settings, signin, signinTotpCode, totp },
       testAccountTracker,
     }) => {
       const credentials = await signInAccount(
         target,
         page,
-        login,
+        settings,
+        signin,
         testAccountTracker
       );
 
-      await settings.goto();
       const { secret } = await addTotp(settings, totp);
       await settings.signOut();
-      await login.login(credentials.email, credentials.password);
-      await login.setTotp(secret);
+      await signin.fillOutEmailFirstForm(credentials.email);
+      await signin.fillOutPasswordForm(credentials.password);
+      await expect(page).toHaveURL(/signin_totp_code/);
+      const code = await getCode(secret);
+      await signinTotpCode.fillOutCodeForm(code);
```

Comment:
I think this set of steps is repeating within this file? Maybe extract to a file level method?